xapian-applications/omega/clickmodel/session.h

```diff
@@ -0,0 +1,57 @@
+/** @file session.h
+ * @brief Session class for handling search session data.
+ */
+/* Copyright (C) 2017 Vivek Pal
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <string>
```

Comment:
Sorry, I missed this before but headers really should have guards to make them safe to include multiple times (the accepted wisdom is to make header inclusion idempotent rather than try to avoid this situation, which is often hard when it happens indirectly) - i.e. `#ifndef OMEGA_INCLUDED_SESSION_H` etc.